File: daa.swift
class daa {
    var first: [Int]
    var second: [Int]
    var iterfirst: Int = 0
    var itersecond: Int = 0
    var sum1: Int = 0
    var sum2: Int = 0

    init(_ first: [Int], _ second: [Int]) {
        self.first = first
        self.second = second
    }

    func solve() -> Int {
        let length = (first.remove(at: 0), second.remove(at: 0))

        while iterfirst < length.0, itersecond < length.1 {
            if first[iterfirst] > second[itersecond] {
                inc_second()
            } else if first[iterfirst] < second[itersecond] {
                inc_first()
            } else {
                inc_first()
                inc_second()
                if sum1 < sum2 {
                    sum1 = sum2
                } else {
                    sum2 = sum1
                }
            }
        }

        while iterfirst < length.0 {
            inc_first()
        }
        while itersecond < length.1 {
            inc_second()
        }

        return max(sum1, sum2)
    }

    func inc_first() {
        sum1 += first[iterfirst]
        iterfirst += 1
    }

    func inc_second() {
        sum2 += second[itersecond]
        itersecond += 1
    }
}

while true {
    let first = readLine()?.split(separator: " ").map { Int(String($0))! }
    if first?[0] == 0 { break }
    let second = readLine()?.split(separator: " ").map { Int(String($0))! }
    print(daa(first!, second!).solve())
}

File: daa.rs
fn input() -> Vec<i64> {
    let mut input = String::new();
    std::io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");
    input
        .trim()
        .to_string()
        .split(' ')
        .map(|s| s.parse().expect("Please type a number!"))
        .collect()
}

fn main() {
    loop {
        let mut num1 = input();
        if num1[0] == 0 {
            break;
        }
        let len1: usize = num1.remove(0) as usize;

        let mut num2 = input();
        let len2: usize = num2.remove(0) as usize;

        let (mut iter1, mut iter2, mut sum1, mut sum2) = (0usize, 0usize, 0i64, 0i64);
        while iter1 < len1 && iter2 < len2 {
            if num1[iter1] > num2[iter2] {
                sum2 += num2[iter2];
                iter2 += 1;
            } else if num1[iter1] < num2[iter2] {
                sum1 += num1[iter1];
                iter1 += 1;
            } else {
                sum2 += num2[iter2];
                sum1 += num1[iter1];
                if sum1 < sum2 {
                    sum1 = sum2
                } else {
                    sum2 = sum1
                }
                iter2 += 1;
                iter1 += 1;
            }
        }
        while iter1 < len1 {
            sum1 += num1[iter1];
            iter1 += 1;
        }
        while iter2 < len2 {
            sum2 += num2[iter2];
            iter2 += 1;
        }
        println!("{}", std::cmp::max(sum1, sum2));
    }
}
